.github/workflows/build-and-deploy.yml

```diff
@@ -42,12 +42,12 @@ jobs:
       - name: Build
         run: PACK_CMD=./pack make build-linux build-wine
       - uses: azure/docker-login@v1
-        if: (github.event_name == 'repository_dispatch') || (github.event_name == 'push' && github.ref == 'refs/heads/main')
```

Comment:
Why remove the dispatch?

Reply:
AFAIK this step should run only on the `main` branch.  The workflow is [executed on repository_dispatch](https://github.com/buildpacks/samples/blob/8202f569ee8823604f28809eb4c930528e9d4844/.github/workflows/build-and-deploy.yml#L10), but I believe the step should only run on the `main` branch.